Zoltan:  non-compliant build without c99

## Bug Report
@trilinos/zoltan

### Description
@iulian787 reports
```
And I did not use -std=c99 or something like that, that could have fixed the error

"../../zoltan/src/params/free_params.c: In function ‘Zoltan_Deserialize_Params’:
../../zoltan/src/params/free_params.c:132:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
   for (int i = 0; i < nParam; i++) {
   ^
../../zoltan/src/params/free_params.c:132:3: note: use option -std=c99 or -std=gnu99 to compile your code
make[2]: *** [free_params.o] Error 1
```

We try to keep zoltan compliant with older C compilers for zoltan-only builds.

--- a/packages/zoltan/src/params/free_params.c
+++ b/packages/zoltan/src/params/free_params.c
@@ -123,13 +123,14 @@ int Zoltan_Deserialize_Params(struct Zoltan_Struct *to, char **buf)
 {
   /* Serialize the parameters */
   char *bufptr = *buf;
+  int i;
 
   /* Unpack number of parameters */
   int nParam = *((int *)bufptr);
   bufptr += sizeof(int);
 
   /* Unpack parameters' (name, value) pairs and set them */
-  for (int i = 0; i < nParam; i++) {
+  for (i = 0; i < nParam; i++) {
     char *pname = bufptr;
     char *pval = bufptr + MAX_PARAM_STRING_LEN;
     Zoltan_Set_Param(to, pname, pval);

--- a/packages/zoltan/test/misc_tests/copyZZ.c
+++ b/packages/zoltan/test/misc_tests/copyZZ.c
@@ -42,10 +42,11 @@ void objMulti(void *data, int ngid, int nlid,
 {
   *ierr = ZOLTAN_OK;
   struct Mesh *mesh = (struct Mesh *) data;
-  for (int i = 0; i < mesh->nMyCoords; i++) {
+  int i, j;
+  for (i = 0; i < mesh->nMyCoords; i++) {
     lid[i*nlid] = i+mesh->myFirstCoord;
     gid[i*ngid] = i+mesh->myFirstCoord; 
-    for (int j = 0; j < wdim; j++) wgt[i*wdim+j] = 1.;
+    for (j = 0; j < wdim; j++) wgt[i*wdim+j] = 1.;
   }
 }
 
@@ -56,8 +57,8 @@ void geomMulti(void *data, int ngid, int nlid, int nobj,
                double *coords, int *ierr)
 {
   struct Mesh *mesh = (struct Mesh *) data;
-
-  for (int i = 0; i < nobj; i++) {
+  int i;
+  for (i = 0; i < nobj; i++) {
     coords[i*ndim]   = mesh->x[lid[i*nlid]];
     coords[i*ndim+1] = mesh->y[lid[i*nlid]];
     coords[i*ndim+2] = mesh->z[lid[i*nlid]];
@@ -202,7 +203,8 @@ int run_test(       /* Test options relevant to serialization, such as... */
 
   int answer[nTwo];
   if (me == 0) {
-    for (int i = 0; i < nTwo; i++) {
+    int i;
+    for (i = 0; i < nTwo; i++) {
       double tmp[3]; tmp[0] = xTwo[i]; tmp[1] = yTwo[i]; tmp[2] = zTwo[i];
       Zoltan_LB_Point_PP_Assign(zz, tmp, NULL, &answer[i]);
       printf("Point (%f %f %f) on part %d\n", 
@@ -214,7 +216,8 @@ int run_test(       /* Test options relevant to serialization, such as... */
   /* Each processor computes answer using new struct newZZ */
 
   int errCnt = 0;
-  for (int i = 0; i < nTwo; i++) {
+  int i;
+  for (i = 0; i < nTwo; i++) {
     int newAnswer;
     double tmp[3]; tmp[0] = xTwo[i]; tmp[1] = yTwo[i]; tmp[2] = zTwo[i];
     Zoltan_LB_Point_PP_Assign(newZZ, tmp, NULL, &newAnswer);

